removal of docker runtime mentions in feature gates page (#31498)
kubernetes/website: content/en/docs/reference/command-line-tools-reference/feature-gates.md

- `APIResponseCompression`: Compress the API responses for `LIST` or `GET` requests.
- `APIServerIdentity`: Assign each API server an ID in a cluster.
- `APIServerTracing`: Add support for distributed tracing in the API server.

> - `Accelerators`: Provided an early form of plugin to enable Nvidia GPU support when using Docker, no longer available. See [Device Plugins](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) for more details. 

Reviewer: ```suggestion
- `Accelerators`: Provided an early form of plugin to enable Nvidia GPU support when using
  Docker Engine; no longer available. See
  [Device Plugins](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/) for
  an alternative.
```

  kubelets on Pod log requests.
- `AnyVolumeDataSource`: Enable use of any custom resource as the `DataSource` of a
  {{< glossary_tooltip text="PVC" term_id="persistent-volume-claim" >}}.

> - `AppArmor`: Enable AppArmor based mandatory access control on Linux nodes when using container runtime.

Reviewer: ```suggestion
- `AppArmor`: Enable use of AppArmor mandatory access control for Pods running on Linux nodes.
```
perhaps?

Does this feature gate require Docker Engine? I assume that it doesn't; if it does, we need to address that.

Author: To my knowledge, AppArmor is concerned mainly with LXD, don't have much knowledge of docker engine requirements.

Author: AppArmor feature should work with other container runtimes other than docker, according to `Container runtime supports AppArmor` point in the [Restrict a Container's Access to Resources with AppArmor](https://kubernetes.io/docs/tutorials/security/apparmor/) documentation.

Reviewer: > `AppArmor`: Enable AppArmor based mandatory access control on Linux nodes when using container runtime.

The wording as it stands doesn't make sense, because nodes _always_ use a container runtime (unless they are misconfigured, which doesn't really count).

Reviewer: How about accepting the suggestion I made earlier @SubhasmitaSw ?

Author: The suggestions look great to me. 

> How about accepting the suggestion I made earlier @SubhasmitaSw ?

Author: > > `AppArmor`: Enable AppArmor based mandatory access control on Linux nodes when using container runtime.
> 
> The wording as it stands doesn't make sense, because nodes _always_ use a container runtime (unless they are misconfigured, which doesn't really count).

then something must be done to atone the dilemma on the documentation part?
